eng/pipelines/coreclr/crossgen2-outerloop.yml

```diff
@@ -74,3 +103,102 @@ jobs:
       crossgen2: true
       displayNameArgs: R2R_CG2
       liveLibrariesBuildConfig: Release
+
+# Build Crossgen2 baselines
+# These are the various crossgen2 targets that are supported, and cover all major
+# significantly different code generators
+- template: /eng/pipelines/common/platform-matrix.yml
+  parameters:
+    jobTemplate: /eng/pipelines/coreclr/templates/crossgen2-comparison-build-job.yml
+    buildConfig: Release
+    platforms:
+    - Linux_arm
+    - Linux_arm64
+    - OSX_x64
+    - Windows_NT_x86
+    - Windows_NT_x64
+    jobParameters:
+      testGroup: outerloop
+      liveLibrariesBuildConfig: Release
+
+# test crossgen target Windows X86
+# This job verifies that 32-bit and 64 bit crossgen2 produces the same binaries,
+# and that cross-os targetting works
+- template: /eng/pipelines/common/platform-matrix.yml
+  parameters:
+    jobTemplate: /eng/pipelines/coreclr/templates/crossgen2-comparison-job.yml
+    buildConfig: Release
+    platforms:
+    - OSX_x64
+    - Windows_NT_x86
+    helixQueueGroup: pr
+    helixQueuesTemplate: /eng/pipelines/coreclr/templates/helix-queues-setup.yml
+    jobParameters:
+      testGroup: outerloop
+      liveLibrariesBuildConfig: Release
+      targetos: Windows_NT
+      targetarch: x86
+
+# test target OSX X64
+# verify that cross OS targetting works
+- template: /eng/pipelines/common/platform-matrix.yml
+  parameters:
+    jobTemplate: /eng/pipelines/coreclr/templates/crossgen2-comparison-job.yml
+    buildConfig: Release
+    platforms:
+    - Windows_NT_x64
+    helixQueueGroup: pr
+    helixQueuesTemplate: /eng/pipelines/coreclr/templates/helix-queues-setup.yml
+    jobParameters:
+      testGroup: outerloop
+      liveLibrariesBuildConfig: Release
+      targetos: OSX
+      targetarch: x64
+
+# test target Windows X64
+# verify that cross OS targetting works
+- template: /eng/pipelines/common/platform-matrix.yml
+  parameters:
+    jobTemplate: /eng/pipelines/coreclr/templates/crossgen2-comparison-job.yml
+    buildConfig: Release
+    platforms:
+    - OSX_x64
```

Comment:
Nit - if your intent here is to just pick the most expedient platform for building something, things have changed recently and Linux is now generally considered "fastest". Most YAML scripts are using <code>CoreClrTestBuildHost</code> as the variable representing "the fastest build platform" but I understand that your requirements are somewhat specific here as a hypothetical future switch-over of CoreClrTestBuildHost to Windows thanks to availability of fast lab HW would negate the cross-OS build assumptions of this particular job matrix.

Reply:
In this case I'm specifically trying to have an OSX test here to go with the other Linux and Windows tests. The more interesting case will come when we add a bit of testing around OSX arm64, but that's for a future change.

Reply:
And of course, Osx has a different python environment than everything else. I'll just get rid of the OSX use here. Its not strictly needed to cover the interesting scenarios, and certainly isn't worth the pain right now. We may need to revisit when we get Osx arm64 in place.